README.md
# web-anotasi
web anotasi twogethergroup


changelog.txt
Version: 4.3.0
  - Updated Bootstrap to version 5.0.1
  - Updated all outdated third party vendor libraries to their latest versions
  - Fixed navigation links focus color

Version: 4.2.0
  - Updated Bootstrap to version 5.0.0 Final
  - Updated all outdated third party vendor libraries to their latest versions

Version: 4.1.0
  - Updated Bootstrap to version 5.0.0-beta3
  - Updated all outdated third party vendor libraries to their latest versions
  - Updated the PHP Email Form to V3.1

Version: 4.0.1
  - Updated Bootstrap to version 5.0.0-beta2
  - Updated all outdated third party vendor libraries to their latest versions

Version: 4.0.0
  - The template does not require jQuery anymore
  - Removed jQuery and all the jQuery plugins
  - The assets/js/main.js was rewritten completely with vanilla Javascript. No more jQuery code
  - Restructured the dev version for better development experience
  - Updated the PHP Email Form to V3.0 - No jQuery dependency. Added attachment support

Version: 3.0.0
  - Initial release with Bootstrap v5.0 Beta 1

Version: 2.2.0
  - Updated Bootstrap to version 4.5.3
  - Updated all outdated third party vendor libraries to their latest versions
  - Updated the PHP Email Form to v2.3
  - Other small fixes and improvements

Version: 2.1.0
  - Updated Bootstrap to version 4.5.0
  - Updated the PHP Email Form library to version 2.0 with reCaptcha support
  - Added portfolio-details.html template
  - Aded inner-page.html tempalte
  - Updated all the third party vendor libraries to their latest versions
  - Other small fixes and improvements
  
Version: 2.0.0
  - The template was rebuilt from scratch with the latest Bootstrap version (4.4.1)
  - Added SMPTP support for the contact form script (Pro)
  - Added NodeJS NPM Development version (Pro unlimited & Membership members)
  
Version: 1.0.0
- Initial Release


assets/js/main.js
/**
* Template Name: Bocor - v4.3.0
* Template URL: https://bootstrapmade.com/bocor-bootstrap-template-nice-animation/
* Author: BootstrapMade.com
* License: https://bootstrapmade.com/license/
*/
(function() {
  "use strict";

  /**
   * Easy selector helper function
   */
  const select = (el, all = false) => {
    el = el.trim()
    if (all) {
      return [...document.querySelectorAll(el)]
    } else {
      return document.querySelector(el)
    }
  }

  /**
   * Easy event listener function
   */
  const on = (type, el, listener, all = false) => {
    let selectEl = select(el, all)
    if (selectEl) {
      if (all) {
        selectEl.forEach(e => e.addEventListener(type, listener))
      } else {
        selectEl.addEventListener(type, listener)
      }
    }
  }

  /**
   * Easy on scroll event listener 
   */
  const onscroll = (el, listener) => {
    el.addEventListener('scroll', listener)
  }

  /**
   * Navbar links active state on scroll
   */
  let navbarlinks = select('#navbar .scrollto', true)
  const navbarlinksActive = () => {
    let position = window.scrollY + 200
    navbarlinks.forEach(navbarlink => {
      if (!navbarlink.hash) return
      let section = select(navbarlink.hash)
      if (!section) return
      if (position >= section.offsetTop && position <= (section.offsetTop + section.offsetHeight)) {
        navbarlink.classList.add('active')
      } else {
        navbarlink.classList.remove('active')
      }
    })
  }
  window.addEventListener('load', navbarlinksActive)
  onscroll(document, navbarlinksActive)

  /**
   * Scrolls to an element with header offset
   */
  const scrollto = (el) => {
    let elementPos = select(el).offsetTop
    window.scrollTo({
      top: elementPos,
      behavior: 'smooth'
    })
  }

  /**
   * Back to top button
   */
  let backtotop = select('.back-to-top')
  if (backtotop) {
    const toggleBacktotop = () => {
      if (window.scrollY > 100) {
        backtotop.classList.add('active')
      } else {
        backtotop.classList.remove('active')
      }
    }
    window.addEventListener('load', toggleBacktotop)
    onscroll(document, toggleBacktotop)
  }

  /**
   * Mobile nav toggle
   */
  on('click', '.mobile-nav-toggle', function(e) {
    select('#navbar').classList.toggle('navbar-mobile')
    this.classList.toggle('bi-list')
    this.classList.toggle('bi-x')
  })

  /**
   * Mobile nav dropdowns activate
   */
  on('click', '.navbar .dropdown > a', function(e) {
    if (select('#navbar').classList.contains('navbar-mobile')) {
      e.preventDefault()
      this.nextElementSibling.classList.toggle('dropdown-active')
    }
  }, true)

  /**
   * Scrool with ofset on links with a class name .scrollto
   */
  on('click', '.scrollto', function(e) {
    if (select(this.hash)) {
      e.preventDefault()

      let navbar = select('#navbar')
      if (navbar.classList.contains('navbar-mobile')) {
        navbar.classList.remove('navbar-mobile')
        let navbarToggle = select('.mobile-nav-toggle')
        navbarToggle.classList.toggle('bi-list')
        navbarToggle.classList.toggle('bi-x')
      }
      scrollto(this.hash)
    }
  }, true)

  /**
   * Scroll with ofset on page load with hash links in the url
   */
  window.addEventListener('load', () => {
    if (window.location.hash) {
      if (select(window.location.hash)) {
        scrollto(window.location.hash)
      }
    }
  });

  /**
   * Porfolio isotope and filter
   */
  // window.addEventListener('load', () => {
  //   let portfolioContainer = select('.portfolio-container');
  //   if (portfolioContainer) {
  //     let portfolioIsotope = new Isotope(portfolioContainer, {
  //       itemSelector: '.portfolio-item',
  //       layoutMode: 'fitRows'
  //     });

  //     let portfolioFilters = select('#portfolio-flters li', true);

  //     on('click', '#portfolio-flters li', function(e) {
  //       e.preventDefault();
  //       portfolioFilters.forEach(function(el) {
  //         el.classList.remove('filter-active');
  //       });
  //       this.classList.add('filter-active');

  //       portfolioIsotope.arrange({
  //         filter: this.getAttribute('data-filter')
  //       });
  //       portfolioIsotope.on('arrangeComplete', function() {
  //         AOS.refresh()
  //       });
  //     }, true);
  //   }

  // });

  /**
   * Initiate portfolio lightbox 
   */
  const portfolioLightbox = GLightbox({
    selector: '.portfolio-lightbox'
  });

  /**
   * Portfolio details slider
   */
  new Swiper('.portfolio-details-slider', {
    speed: 400,
    autoplay: {
      delay: 5000,
      disableOnInteraction: false
    },
    pagination: {
      el: '.swiper-pagination',
      type: 'bullets',
      clickable: true
    }
  });

  /**
   * Animation on scroll
   */
  window.addEventListener('load', () => {
    AOS.init({
      duration: 1000,
      easing: "ease-in-out",
      once: true,
      mirror: false
    });
  });

})()

channel.php
<?php

$ch = $_GET['ch'];
if(!isset($_GET['ch']) || $ch < 0 || $ch >= 100){
  header("location: index.html");
  exit();
}

?>

<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta content="width=device-width, initial-scale=1.0" name="viewport">

  <title>2GetherGroup - Channel</title>
  <meta content="" name="description">
  <meta content="" name="keywords">

  <!-- Favicons -->
  <link href="assets/img/favicon.png" rel="icon">
  <link href="assets/img/apple-touch-icon.png" rel="apple-touch-icon">

  <!-- Google Fonts -->
  <link href="https://fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i|Raleway:300,300i,400,400i,500,500i,600,600i,700,700i|Poppins:300,300i,400,400i,500,500i,600,600i,700,700i" rel="stylesheet">

  <script src="https://code.jquery.com/jquery-3.5.1.js"></script>

  <!-- Vendor CSS Files -->
  <link href="assets/vendor/aos/aos.css" rel="stylesheet">
  <link href="assets/vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">
  <link href="assets/vendor/bootstrap-icons/bootstrap-icons.css" rel="stylesheet">
  <link href="assets/vendor/boxicons/css/boxicons.min.css" rel="stylesheet">
  <link href="assets/vendor/glightbox/css/glightbox.min.css" rel="stylesheet">
  <link href="assets/vendor/swiper/swiper-bundle.min.css" rel="stylesheet">
  <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>

  <!-- Template Main CSS File -->
  <link href="assets/css/style.css" rel="stylesheet">
  <script type="text/javascript">
    const flt_cod = ["prank", "skit", "standup", "food", "daily", "review", "reaction", "podcast", "music", "clip", "reality", "game", "otomotive", "technology", "beauty", "education", "horror", "howto", "magic", "challenge", "discussion", "animation", "news"];
    const flt_str = ["Prank", "Comedy Skit", "Stand Up Comedy", "Food Vlog", "Daily Vlog", "Review", "Reaction", "Podcast", "Music", "Cuplikan Film", "Reality Show", "Gaming", "Otomotif", "Teknologi", "Beauty Vlog", "Edukasi", "Horor", "Tutorial", "Magic", "Challenge", "Diskusi", "Animasi", "Berita"];

    $(function(){
      $.get('datafinal.json',function(obj){
        var str = "";
        var desc = "";
        var flt = "<li data-filter='*' class='filter-active'>All</li>";
        obj = obj[<?php echo $ch; ?>];  
        
        obj.category.forEach((categ) => {
          var fltint = -1;
          for(let i = 0; i < flt_cod.length; i++){
            if(flt_cod[i].localeCompare(categ) == 0){
              fltint = i;
              break;
            }
          }
          flt += "<li data-filter='.filter-" + categ + "'>" + flt_str[fltint] + "</li>";
        });

        var channel_id = obj.channel_id;
        var channel_name = obj.channel_name;
        var channel_photo = obj.channel_photo;
        var subs_string = "" + obj.subs_count + " juta";
        var video_count = obj.video_count;
        var videos = obj.videos;

        var yt_link = "https://youtube.com/";
        if(channel_id.startsWith("UC")){
          yt_link += "channel/" + channel_id;
        } else {
          yt_link += "user/" + channel_id;
        }
        
        desc += "<h2 data-aos='fade-in'>" + channel_name + "</h2>";
        desc += "<a href='" + yt_link + "' class='channel-photo'><img src='"+ channel_photo +"' width='20%' height='auto' alt=''></a>";
        desc += "</br></br><p data-aos='fade-in'>" + subs_string + " subscribers.</p>"
        desc += "<p data-aos='fade-in'>" + video_count + " videos.</p>"
        
        $.each(videos,function(n,data){
          str += "<div class='offset-lg-4 col-lg-4 col-md-6 portfolio-item filter-" + data.category + "'>"

          str += "<div class='portfolio-wrap'>";
          
          str+= "<img src='"+ data.thumbnail +"' width='100%' height='auto' class='img-fluid' alt=''>";

          str += "<div class='portfolio-links'>";

          str += "<a href='"+ data.thumbnail +"' data-gallery='portfolioGallery' class='portfolio-lightbox' title='"+ data.title +"'><i class='bi bi-plus'></i></a>";

          str += "<a href='" + data.url + "' title='Watch'><i class='bi bi-link'></i></a> </div> <div class='portfolio-info'> <h4>"+ data.title +"</h4> </div> </div> </div>"
          
        });
        $('#portfolio-flters').html(flt);
        $('.portfolio-container').html(str);
        $('#channel-desc').html(desc);
        doIsotope();
      });
    });

    const select = (el, all = false) => {
      el = el.trim()
      if (all) {
        return [...document.querySelectorAll(el)]
      } else {
        return document.querySelector(el)
      }
    }

    const on = (type, el, listener, all = false) => {
      let selectEl = select(el, all)
      if (selectEl) {
        if (all) {
          selectEl.forEach(e => e.addEventListener(type, listener))
        } else {
          selectEl.addEventListener(type, listener)
        }
      }
    }

    function doIsotope(){
      let portfolioContainer = select('.portfolio-container');
      if (portfolioContainer) {
        let portfolioIsotope = new Isotope(portfolioContainer, {
          itemSelector: '.portfolio-item',
          layoutMode: 'vertical'
        });

        let portfolioFilters = select('#portfolio-flters li', true);

        on('click', '#portfolio-flters li', function(e) {
          e.preventDefault();
          portfolioFilters.forEach(function(el) {
            el.classList.remove('filter-active');
          });
          this.classList.add('filter-active');

          portfolioIsotope.arrange({
            filter: this.getAttribute('data-filter')
          });
          portfolioIsotope.on('arrangeComplete', function() {
            AOS.refresh()
          });
        }, true);
      }
      setTimeout(function(){select('.filter-active').click();}, 3000);
    }
  </script>

<script type="text/javascript">

// Load the Visualization API and the corechart package.
google.charts.load('current', {'packages':['corechart']});

// Set a callback to run when the Google Visualization API is loaded.
google.charts.setOnLoadCallback(drawChart);

// Callback that creates and populates a data table,
// instantiates the pie chart, passes in the data and
// draws it.
function drawChart() {
  var ch = <?php echo $ch; ?>;
  var data = new google.visualization.DataTable();
  var request = new XMLHttpRequest();
  request.open("GET", "./datafinal.json", false);
  request.send(null)
  var obj = JSON.parse(request.responseText);
  obj = obj[ch];

  data.addColumn('string', 'Category');
  data.addColumn('number', 'Channel Amount');
  var category_count = [];
  for(let i = 0; i < flt_cod.length; i++){
    category_count[i] = 0;
  }

  //get all categories
  for(let j = 0; j < obj.videos.length; j++){
    
    //compare to list categories
    for(let i = 0; i < flt_cod.length; i++){
      if(flt_cod[i].localeCompare(obj.videos[j].category) == 0){
        //add increment to category_count
        category_count[i]++;
        break;
      }
    }
  }

  for(let i = 0; i < flt_str.length; i++){
    data.addRow([flt_str[i], category_count[i]]);
  }

  // Set chart options
  var options = {'title':'Category Per Video in this channel\'s content.',
                 'titleTextStyle':{
                    fontSize: 17
                 },
                 'height':400,
                 'legend':{'position':'none'},
                 'animation':{
                     'duration': 1000,
                     'startup': true,
                     'easing':'inAndOut'
                 },
                 'tooltip':{
                     'trigger':'selection'
                 }};

  // Instantiate and draw our chart, passing in some options.
  var chart = new google.visualization.PieChart(document.getElementById('pie_chart'));
  chart.draw(data, options);
}
</script>

  <!-- =======================================================
  * Template Name: Bocor - v4.3.0
  * Template URL: https://bootstrapmade.com/bocor-bootstrap-template-nice-animation/
  * Author: BootstrapMade.com
  * License: https://bootstrapmade.com/license/
  ======================================================== -->
  
</head>

<body>

  <!-- ======= Header ======= -->
  <header id="header">
    <div class="container d-flex align-items-center justify-content-between">

      <div class="logo">
        <h1><a href="index.html">2GetherGroup<span>.</span></a></h1>
        <!-- Uncomment below if you prefer to use an image logo -->
        <!-- <a href="index.html"><img src="assets/img/logo.png" alt="" class="img-fluid"></a> -->
      </div>

      <nav id="navbar" class="navbar">
        <ul>
          <li><a class="nav-link scrollto active" href="./index.html">Home</a></li>
          <li><a class="nav-link scrollto" href="./index.html#team">Team</a></li>
          <li><a class="nav-link scrollto" href="./index.html#features">Features</a></li>
          <li><a class="nav-link scrollto" href="./index.html#graph">Chart</a></li>
          <li><a class="nav-link scrollto" href="./index.html#portfolio">Channel</a></li>
        </ul>
        <i class="bi bi-list mobile-nav-toggle"></i>
      </nav><!-- .navbar -->

    </div>
  </header><!-- End Header -->

  <!-- ======= Hero Section ======= -->
  <section id="hero">

    <div class="container">
      <div class="row d-flex align-items-center">
        <div class="offset-lg-3 col-lg-6 order-1 order-lg-2 hero-img" data-aos="fade-up">
          <img src="assets/img/Logo polos.png" class="img-fluid" alt="">
        </div>
        <a></a>
      </div>
    </div>

  </section><!-- End Hero -->

  <main id="main">
    <!-- ======= Portfolio Section ======= -->
    <section id="portfolio" class="portfolio section-bg">
      <div class="container">

        <div id="channel-desc" class="section-title" data-aos="fade-up"></div>
        <div id="pie_chart" class="offset-lg-3 col-lg-6" data-aos="fade-up"></div>

        <div class="row">
          <div class="col-lg-12">
            <ul id="portfolio-flters">
            </ul>
          </div>
        </div>

        <div class="row portfolio-container" data-aos="fade-up"></div>

      </div>
    </section><!-- End Portfolio Section -->

  </main><!-- End #main -->

  <!-- ======= Footer ======= -->
  <footer id="footer">

    <div class="footer-top">

      <div class="container">

        <div class="row  justify-content-center">
          <div class="col-lg-6">
            <h3>2GETHERGROUP</h3>
          </div>
        </div>

        <div class="social-links">
          <a href="https://github.com/twogethergroup/" class="github"><i class="bx bxl-github"></i></a>
        </div>

      </div>
    </div>

    <div class="container footer-bottom clearfix">
      <div class="copyright">
        &copy; Copyright <strong><span>Bocor</span></strong>. Modified by 2GetherGroup
      </div>
      <div class="credits">
        <!-- All the links in the footer should remain intact. -->
        <!-- You can delete the links only if you purchased the pro version. -->
        <!-- Licensing information: https://bootstrapmade.com/license/ -->
        <!-- Purchase the pro version with working PHP/AJAX contact form: https://bootstrapmade.com/bocor-bootstrap-template-nice-animation/ -->
          Modified by <a href="https://github.com/twogethergroup/">TwoGetherGroup</a>
      </div>
    </div>
  </footer><!-- End Footer -->

  <a href="#" class="back-to-top d-flex align-items-center justify-content-center"><i class="bi bi-arrow-up-short"></i></a>

  <!-- Vendor JS Files -->
  <script src="assets/vendor/aos/aos.js"></script>
  <script src="assets/vendor/bootstrap/js/bootstrap.bundle.min.js"></script>
  <script src="assets/vendor/glightbox/js/glightbox.min.js"></script>
  <script src="assets/vendor/isotope-layout/isotope.pkgd.min.js"></script>
  <script src="assets/vendor/php-email-form/validate.js"></script>
  <script src="assets/vendor/swiper/swiper-bundle.min.js"></script>

  <!-- Template Main JS File -->
  <script src="assets/js/main.js"></script> 
</body>

</html>